src/map.cpp

```diff
@@ -1503,7 +1504,9 @@ int map::bash_rating(const int str, const int x, const int y)
         return 10;
     }
 
-    return (10 * (str - bash_min)) / (bash_max - bash_min);
+    int ret = (10 * (str - bash_min)) / (bash_max - bash_min);
+    // Round up to 1, so that desperate NPCs can try to bash down walls
+    return ret > 0 ? ret : 1;
```

Comment:
In my opinion `return std::max( ret, 1 );` is much easier to read and understand.